modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java

```diff
@@ -252,4 +281,131 @@ private void errorsCheck(SCIMResponse response) {
+     * This method will replace the bulk id with the created user id.
+     *
+     * @param bulkRequestContent   Bulk request content.
+     * @param userIDMappings       User id bulk id mapping.
+     * @param method               HTTP method.
+     * @throws BadRequestException Bad request exception.
+     */
+    private void processBulkRequestContent(BulkRequestContent bulkRequestContent, Map<String, String> userIDMappings,
+                                           String method) throws BadRequestException {
+
+        try {
+            if (userIDMappings == null || userIDMappings.isEmpty()
+                    || method.equals(SCIMConstants.OperationalConstants.DELETE)) {
+                return;
+            }
+            // Parse the data field to a JSON object.
+            JSONObject dataJson = new JSONObject(bulkRequestContent.getData());
+            String usersOrMembersKey = getUsersOrMembersKey(bulkRequestContent.getPath());
+            JSONArray usersArray = getUserArray(dataJson, method, usersOrMembersKey);
+
+            if (usersArray != null) {
+                String bulkIdPrefix = SCIMConstants.OperationalConstants.BULK_ID + ":";
+                for (int i = 0; i < usersArray.length(); i++) {
+                    JSONObject user = usersArray.getJSONObject(i);
+                    String userValue = user.getString(SCIMConstants.OperationalConstants.VALUE);
+                    if (userValue.startsWith(bulkIdPrefix)) {
+                        String userBulkId = userValue.substring(bulkIdPrefix.length());
+                        String userId = userIDMappings.get(userBulkId);
+                        if (StringUtils.isNotBlank(userId)) {
+                            user.put(SCIMConstants.OperationalConstants.VALUE, userId);
+                        }
+                    }
+                }
+            }
+            bulkRequestContent.setData(dataJson.toString());
+        } catch (JSONException e) {
+            throw new BadRequestException("Error while parsing the data field of the bulk request content",
+                    ResponseCodeConstants.INVALID_SYNTAX);
+        }
+    }
+
+    private String getUsersOrMembersKey(String path) {
+
+        return path.contains(SCIMConstants.ROLE_ENDPOINT) ? SCIMConstants.RoleSchemaConstants.USERS :
+                SCIMConstants.GroupSchemaConstants.MEMBERS;
+    }
+
+    /**
+     * This method is used to get the user array from the data JSON object.
+     *
+     * @param dataJson          SCIM data JSON object.
+     * @param method            HTTP method.
+     * @param usersOrMembersKey Users or members key.
+     * @return User array
+     * @throws JSONException    JSON exception.
+     */
+    private JSONArray getUserArray(JSONObject dataJson, String method, String usersOrMembersKey)
+            throws JSONException {
+
+        switch (method) {
+            case SCIMConstants.OperationalConstants.POST:
```

Comment:
If there is no specific logic for POST, can't we fall it under default behavior?